### PR TITLE
feat: add modal component

### PR DESCRIPTION
**Pull request by ngochuyduong**

### CODE REVIEW

**Review by loc-nguyenthien — APPROVED** [review]

File: src/Modal/Modal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export const ModalComponent: ComponentStory<typeof Modal> = () => {
+  const [visible, setVisible] = React.useState(false)
+
+  const showModal = useCallback(() => setVisible(true), [])
+  const hideModal = useCallback(() => setVisible(false), [])
+
+  return (
+    <View style={styles.viewContainer}>
+      <Modal visible={visible} onClose={hideModal} contentContainerStyle={styles.contentContainerStyle}>
+        <View>
```
Comment (huydosgtech) [inline src/Modal/Modal.stories.tsx:20]: Can we use a props {children} in modal ?

**Review by huydosgtech — CHANGES_REQUESTED** [review]

File: src/Modal/Modal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  )
+}
+
+const styles = StyleSheet.create({
+  viewContainer: {
+    flex: 1,
+  },
+  contentContainerStyle: {
+    width: 300,
+    height: 300,
```
Comment (huydosgtech) [inline src/Modal/Modal.stories.tsx:38]: It's not responsive layout

Comment (ngochuyduong) [reply src/Modal/Modal.stories.tsx:38]: This is just a storyboard file for testing. Can check the Modal.tsx file

File: src/Modal/Modal.stories.tsx (hunk 1 above)
Comment (ngochuyduong) [reply src/Modal/Modal.stories.tsx:20]: This is just a storyboard file for testing. Can check the Modal.tsx file

File: src/Modal/Modal.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      </Modal>
+      <View style={styles.buttonContainer}>
+        <Button onPress={showModal} title="Show" />
+      </View>
+    </View>
+  )
+}
+
+const styles = StyleSheet.create({
+  viewContainer: {
```
Comment (TranQuangPhi) [inline src/Modal/Modal.stories.tsx:33]: Please use styled components. This point has been added to the coding convention
https://trello.com/c/ClAU12Cu/43-coding-convention

File: src/Modal/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  ) => {
+    return (
+      <Modal
+        ref={ref}
+        visible={visible}
+        transparent
+        animationType={animationType}
+        onRequestClose={() => (dismissable ? onClose && onClose() : undefined)}>
+        <Pressable
+          style={StyleSheet.flatten([styles.backdrop, {backgroundColor: backdropColor}])}
```
Comment (ThinhKimVo) [inline src/Modal/Modal.tsx:60]: what is different between stylesheet compose and flatten?

File: src/Modal/Modal.stories.tsx (hunk 3 above)
Comment (ngochuyduong) [reply src/Modal/Modal.stories.tsx:33]: will update later

File: src/Modal/Modal.tsx (hunk 4 above)
Comment (ngochuyduong) [reply src/Modal/Modal.tsx:60]: Compose: Combines two styles such that style2 will override any styles in style1. If either style is falsy, the other one is returned without allocating an array, saving allocations and maintaining reference equality for PureComponent checks.

Flatten: Flattens an array of style objects, into one aggregated style object.

Actually we will update according to our convention

**Pull request closed without merging**